Kinetic_surface_reconstruction/include/CGAL/Kinetic_surface_reconstruction_3.h

```diff
@@ -0,0 +1,2003 @@
+// Copyright (c) 2023 GeometryFactory Sarl (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sven Oesau, Florent Lafarge, Dmitry Anisimov, Simon Giraudot
+
+#ifndef CGAL_KINETIC_SURFACE_RECONSTRUCTION_3_H
+#define CGAL_KINETIC_SURFACE_RECONSTRUCTION_3_H
+
+#include <CGAL/license/Kinetic_shape_partition.h>
+
+#include <CGAL/Kinetic_shape_partition_3.h>
+#include <CGAL/KSP_3/Graphcut.h>
+
+#include <CGAL/IO/PLY.h>
+#include <CGAL/Point_set_3.h>
+#include <CGAL/Point_set_3/IO.h>
+#include <CGAL/Shape_detection/Region_growing/Region_growing.h>
+#include <CGAL/Shape_detection/Region_growing/Point_set.h>
+#include <CGAL/Delaunay_triangulation_2.h>
+#include <CGAL/Delaunay_triangulation_3.h>
+#include <CGAL/Triangulation_face_base_with_info_2.h>
+#include <CGAL/Triangulation_vertex_base_with_info_2.h>
+#include <CGAL/KSP/debug.h>
+#include <CGAL/Shape_regularization/regularize_planes.h>
+#include <CGAL/bounding_box.h>
+
+#include <boost/range/adaptor/transformed.hpp>
+
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/boost/graph/helpers.h>
+
+namespace CGAL
+{
+/*!
+* \ingroup PkgKineticSurfaceReconstructionRef
+  \brief Reconstruction pipeline for piece-wise planar surface reconstruction from a point cloud via inside/outside labeling of a kinetic partition using graph cut.
+
+  \tparam GeomTraits
+    must be a model of `KineticShapePartitionTraits_3`.
+
+  \tparam PointSet
+    must be a range of 3D points and corresponding 3D normal vectors whose iterator type is `RandomAccessIterator`.
+
+  \tparam PointMap
+    a model of `ReadablePropertyMap` whose key type is the value type of the `PointSet` and value type is `GeomTraits::Point_3`
+
+  \tparam NormalMap
+    a model of `ReadablePropertyMap` whose key type is the value type of the `PointSet` and value type is `GeomTraits::Vector_3`
+
+  \tparam IntersectionKernel
+    must be a model of `Kernel` using exact computations. Defaults to `CGAL::Exact_predicates_exact_constructions_kernel`. Used for the internal kinetic shape partition.
+*/
+template<typename GeomTraits, typename PointSet, typename PointMap, typename NormalMap, typename IntersectionKernel = CGAL::Exact_predicates_exact_constructions_kernel>
+class Kinetic_surface_reconstruction_3 {
+public:
+  using Kernel = GeomTraits;
+  using Intersection_kernel = IntersectionKernel;
+
+  using FT = typename Kernel::FT;
+  using Point_3 = typename Kernel::Point_3;
+  using Plane_3 = typename Kernel::Plane_3;
+  using Point_set = PointSet;
+
+  using KSP = Kinetic_shape_partition_3<Kernel, Intersection_kernel>;
+
+  using Point_map = PointMap;
+  using Normal_map = NormalMap;
+
+  /*!
+  \brief Creates a Kinetic_shape_reconstruction_3 object.
```

Comment:
comments of member functions start lower case if that makes a sentence with the member function